src/vm.rs

```diff
@@ -73,6 +75,34 @@ impl MmioOpcode {
     }
 }
 
+#[test_case]
+fn MmioOpcodeTest() -> TestResult {
```

Comment:
is the purpose of the `TestResult` return type to avoid crashing the runner when a unit test panics, e.g. when an `assert!` fires? If so, should we be returning `Fail` instead of asserting below?

Reply:
fix

Reply:
Still?